support/gin/gin.go

```diff
@@ -0,0 +1,49 @@
+package gin
+
+import (
+	"net/http"
+
+	"github.com/davidebianchi/gswagger/apirouter"
+	
+	"github.com/gin-gonic/gin"
+)
+
+// HandlerFunc is the http type handler used by Gin
+type HandlerFunc gin.HandlerFunc
+type Route = gin.IRoutes
+
+type ginRouter struct {
+	router *gin.Engine
+}
+
+func (r ginRouter) AddRoute(method string, path string, handler HandlerFunc) Route {
+	switch method {
+	case http.MethodGet:
+		return r.router.GET(path, gin.HandlerFunc(handler))
+	case http.MethodPost:
+		return r.router.POST(path, gin.HandlerFunc(handler))
+	case http.MethodPut:
+		return r.router.PUT(path, gin.HandlerFunc(handler))
+	case http.MethodDelete:
+		return r.router.DELETE(path, gin.HandlerFunc(handler))
+	default:
+		return r.router.Any(path, gin.HandlerFunc(handler))
+	}
```

Comment:
This switch/case can lead to unexpected behaviour (for example, if set PATCH method this will set Any methods, which are [GET, POST, PUT, PATCH, HEAD, OPTIONS, DELETE, CONNECT, TRACE](https://pkg.go.dev/github.com/gin-gonic/gin#RouterGroup.Any)).

What do you think about replace it with the [match method](https://pkg.go.dev/github.com/gin-gonic/gin#RouterGroup.Match)? 
Something like:

```suggestion
	return r.router.Match([]string{method}, path, handler)
```